client/extensions/woocommerce/app/settings/email/index.js

```diff
@@ -5,17 +5,23 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { localize } from 'i18n-calypso';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import Main from 'components/main';
+import Button from 'components/button';
 import MailChimp from './mailchimp';
 import ActionHeader from 'woocommerce/components/action-header';
 import SettingsNavigation from '../navigation';
 import { getLink } from 'woocommerce/lib/nav-utils';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { mailChimpSaveSettings } from 'woocommerce/state/sites/settings/email/actions';
+import { isSavingSettings } from 'woocommerce/state/sites/settings/email/selectors';
 
-const SettingsEmail = ( { site, siteId, translate, className, params } ) => {
+const SettingsEmail = ( { site, siteId, translate, className, params, isSaving, mailChimpSaveSettings: saveSettings } ) => {
```

Comment:
Instead of having `SettingsEmail` responsible for saving each child component's settings, it could be better to have `SettingsEmail` ask the child component to save itself (the child component probably knows best what needs to be saved, if saving is supported at this time, etc etc) To this end, could we use instance methods? https://www.andrewhfarmer.com/component-communication/

We could also use a prop-passed callback to allow the child to let the parent know that something bad happened during saving, or if forms are dirty, etc etc

Might be a good thing to discuss as a group

Reply:
This is exactly what is happening here. `saveSettings` has no knowledge of data. This is is just an action that indicates that user clicked Save button. Components can subscribe to this action via store and the `isSavingSettings` selector and respond with appropriate action. Currently, the only component that supports this behavior is `Settings` in the dashboard and this is only one that needs this - all other settings are saved during various wizard steps - with other save button.

Instance methods are problematic. If you have a big nested structure you need a long chain of references. Here you would need SettingsEmail->MailChimp->MailChimpDasboard->Settings in all these components you would need function for `ref`

We can still discuss this - I don't mind :)

Reply:
Besides that. Ref does not work with hoc like localize.

Reply:
I see your points - and yep, boo: https://reactjs.org/docs/higher-order-components.html#refs-arent-passed-through